ebpfapi/rpc_client.cpp

```diff
@@ -50,9 +60,22 @@ ebpf_rpc_load_program(
         return result;
 }
 
-RPC_STATUS
-initialize_rpc_binding()
+/**
+ * @brief Initialize the RPC binding handle. This is expensive and should be
+ * done once when required. This function is idempotent and thread safe.
+ *
+ * @retval RPC_S_OK The binding handle was initialized successfully.
+ * @retval RPC_S_* The binding handle could not be initialized.
+ */
+static RPC_STATUS
+_initialize_rpc_binding()
 {
+    std::unique_lock lock(_rpc_binding_handle_mutex);
```

Comment:
nit: Maybe put all the contents of this function under 
```#if !defined(CONFIG_BPF_JIT_DISABLED) || !defined(CONFIG_BPF_INTERPRETER_DISABLED)```

Reply:
I think this might be discarded by the linker already if the code isn't reachable from exported DLL entry points.